spec/unit/configurer_spec.rb

```diff
@@ -690,7 +690,8 @@ def expects_neither_new_or_cached_catalog
         @agent.instance_variable_set(:@environment, 'second_env')
         expects_new_catalog_only(@catalog)
 
-        Puppet.expects(:err).with(regexp_matches(/Could not/)).times(3)
+        @agent.stubs(:send_report)
+        @agent.stubs(:convert_catalog).returns(@catalog)
```

Comment:
@joshcooper ah, makes sense. `to_ral` didn't do it, though. It seems like the only way to avoid the class and resource file errors is to stub out `@catalog.write_class_file` and `@catalog.write_resource_file`